### DISCUSSION TITLE
Bad Private Key error on Google Auth

### DISCUSSION THREAD

**Discussion by jeremymoore** [Web3Auth/Web3Auth#1330] [Help]
We're getting a spinning blue wheel on the "Constructing your key on..." open login popup. 

While debugging we see `Error: Bad private key`. 

![image](https://user-images.githubusercontent.com/1245814/219248762-55456468-ee89-430c-887d-915320be8df8.png)

We're implementing on cyan using Web3AuthCore with social auth providers like google.

**Comment by shahbaz17** [comment]
Hey @jeremymoore 

Could you share the web3auth initialization and login code here? Also, please share the SDK version here.

**Reply by jeremymoore** [reply]
I've extracted this and it should be pretty close

```
const adapter = new OpenloginAdapter({
  adapterSettings: {
    network: 'cyan',
    clientId: web3AuthClientId,
    uxMode: 'redirect',
    redirectUrl: redirectUrl,
    loginConfig: {
      google: {
        name: 'Google',
        verifier: 'google-verifier',
        typeOfLogin: 'google',
        clientId: adapterIds.google,
      }
    },
  },
})

const web3authCore = new Web3AuthCore({
  chainConfig: {
    chainNamespace: 'eip155',
    chainId: '0x1',
    rpcTarget: 'https://eth-mainnet.g.alchemy.com/v2/....',
    blockExplorer: 'https://etherscan.io/',
    ticker: 'ETH',
    tickerName: 'Ethereum',
  },
  clientId: web3AuthClientId,
})

subscribeAuthEvents(web3authCore)

if (web3authCore.status !== 'connected') {
  web3authCore.configureAdapter(adapter)
  await web3authCore.init()
}

const login = async () => {

  // ....

  await web3auth.connectTo(adapter.name, {
    loginProvider: connectMethod,
  })

  // ...

}

**Reply by jeremymoore** [reply]
```
"@web3auth/core": "^4.0.0",
"@web3auth/openlogin-adapter": "^4.0.0",
"@web3auth/web3auth": "^2.1.3",

**Reply by shahbaz17** [reply]
Can you update the following to the latest?

```js
"@web3auth/core": "^4.6.0",
"@web3auth/openlogin-adapter": "^4.6.0",
```

Also, Is `'google-verifier'` used in the verifier field created by you? Because I don't see this verifier on the cyan network.

**Reply by jeremymoore** [reply]
i replaced the verifier name for the purposes of the post... i'll try updating the sdk. Do you expect there to have been breaking changes  recently?

**Reply by shahbaz17** [reply]
If you're to use google verifier for custom authentication then please follow https://web3auth.io/docs/custom-authentication/social-providers/google and pass the created verifier name.

But before moving forward. May I know what are you trying to use? 

A google login by web3auth or google login with your config without Modal?

**Reply by yashovardhan** [reply]
Also can you remove the package @web3auth/web3auth - it has been deprecated.

**Reply by jeremymoore** [reply]
google login with your config without Modal

**Reply by jeremymoore** [reply]
We do have the implementation following your link, I think we're good there unless there's something specific you see a problem with.

**Reply by shahbaz17** [reply]
Please use the below config, it will work:

```js
const openloginAdapter = new OpenloginAdapter({
  adapterSettings: {
    loginConfig: {
      google: {
        verifier: "tronic-google",
        typeOfLogin: "google",
        clientId: "GOOGLE_CLIENT_ID", //use your app client id you got from google
      },
    },
  },
});
```

**Comment by chaitanyapotti** [comment]
Thanks for reporting this. I've fixed the issue and pushed a new build.
Although, this should not occur frequently.! It's for a rare case where a generated sessionId is not padded

**Comment by jeremymoore** [comment]
@chaitanyapotti once you made this update we no longer see the Bad Private Key error, however we immediately began seeing the following: 

>Failed to connect with wallet. Wallet adapter is not ready yet, Please wait for init function to resolve before calling connect/connectTo function

On further investigation, this is not an issue in Testnet. 

We ran the sample app, and are seeing the same success in the testnet, and failure in mainnet.

**Comment by jeremymoore** [comment]
We create new verifiers and the issue is resolved. Can you help us understand why the previous verifiers suddenly stopped working?

**Comment by chaitanyapotti** [comment]
Old verifiers should also be working as intended. Pls check again
